### ISSUE TITLE
undefined symbol: MapViewOfFile2

### ISSUE THREAD

**Issue by zux0x3a** [ziglang/zig#22040]
### Zig Version

0.13.0

### Steps to Reproduce and Observed Output

it seems MapViewOfFile2 is not declared or linked into Zig code base yet, currently i am declaring it externally : 

```zig
pub extern "kernel32" fn MapViewOfFile2(
    FileMappingHandle: HANDLE,
    ProcessHandle: HANDLE,
    Offset: u64, 
    BaseAddress: ?*anyopaque, 
    ViewSize: usize, // SIZE_T
    AllocationType: u32, // ULONG
    PageProtection: u32, // ULONG
) callconv(WINAPI) ?*anyopaque; 
```

raising the following linking error : 

```
error: warning(link): unexpected LLD stderr:
lld-link: warning: undefined symbol: MapViewOfFile2
```

any alternative solution? if we imported the C headeand then call it from it's name space should be a temporary solution? 

### Expected Output

error: warning(link): unexpected LLD stderr:
lld-link: warning: undefined symbol: MapViewOfFile2

**Comment by lawrenceamer** [comment]
it seems it is not part of Mingw64 https://github.com/msys2-contrib/mingw-w64/blob/master/mingw-w64-headers/include/memoryapi.h

**Comment by nektro** [comment]
msdn https://learn.microsoft.com/en-us/windows/win32/api/memoryapi/nf-memoryapi-mapviewoffile2

**Comment by lawrenceamer** [comment]
@nektro I think you are referring to the following 

```
pub extern "api-ms-win-core-memory-l1-1-5" fn UnmapViewOfFile2(
    Process: ?HANDLE,
    BaseAddress: ?*anyopaque,
    UnmapFlags: UNMAP_VIEW_OF_FILE_FLAGS,
) callconv(@import("std").os.windows.WINAPI) BOOL;
```

and thats limited to windows10.0.15063 check out here : https://github.com/marlersoft/zigwin32/blob/32d085ee67374ad2ec24fc012e6876ca27958fb7/win32/system/memory.zig#L1098

**Comment by mesidex** [comment]
As a workaround, does NtMapViewOfSection suit your use case?

[https://github.com/ziglang/zig/blob/e163ae794b6d6598833c6a8248a818e1e49ce794/lib/std/os/windows/ntdll.zig#L144](https://github.com/ziglang/zig/blob/e163ae794b6d6598833c6a8248a818e1e49ce794/lib/std/os/windows/ntdll.zig#L144)

**Comment by vtorri** [comment]
I have it:

```
$ fgrep -r MapViewOfFile2 /mingw64/include/memoryapi.h
  WINBASEAPI PVOID MapViewOfFile2(HANDLE FileMappingHandle, HANDLE ProcessHandle, ULONG64 Offset, PVOID BaseAddress, SIZE_T ViewSize, ULONG AllocationType, ULONG PageProtection);
```

**Comment by alexrp** [comment]
The problem is that it isn't present in any of MinGW-w64's library definition files. Someone needs to file a bug or send a patch upstream.

**Comment by vtorri** [comment]
I use mingw since around 2008, and there is no problem with this function. mingw-w64 provides the header files and the import libraries. The DLL is in the system. I used it for example in the EFL and I compile on Windows and cross compile on linux without any problem

**Comment by alexrp** [comment]
The only occurrence of `MapViewOfFile2` in the MinGW-w64 source tree is in `mingw-w64-headers/include/memoryapi.h`. I don't see it mentioned in any of the import libraries.

**Comment by alexrp** [comment]
Ah, I wasn't aware of that detail. Then this still looks like a MinGW-w64 bug, because it declares `MapViewOfFile2` in `memoryapi.h` but does not define it anywhere.

**Comment by vtorri** [comment]
that's how it is defined in the visual studio SDK. I've just reported this to the mingw-w64 devs and one the mingw-w64 devs told me that he will fix it soon

**Comment by zux0x3a** [comment]
@alexrp @vtorri I don't think it is an issue with mingw-w64 while it is working perfectly if you depend on cc compiler. the issue that there is no binding for this specific API call because it is inline function which i fixed in previous extern inline call. it can be fixed by ziglang and not in mingw64

**Comment by vtorri** [comment]
https://github.com/mingw-w64/mingw-w64/commit/c67e9a7ab6d4551c65f7d81dccbd8826e4b5ad78

at least, other people who need MapViewOfFile2() will be able to use it
